python/pyarrow/lib.pxd

```diff
@@ -519,6 +519,7 @@ cdef class Table(_Tabular):
         CTable* table
 
     cdef void init(self, const shared_ptr[CTable]& table)
+    cdef void _assert_cpu(self) except *
```

Comment:
Now you added this to `_Tabular`, it might no longer be needed to repeat this here?

Reply:
I thought so too, but I get a runtime error if I leave it out. Since its a `cdef` function, cython adds it to a vtable for that class. I guess Cython needs the parent class to have an explicitly defined vtable entry?

```
>   self._assert_cpu()
E   AttributeError: 'pyarrow.lib.RecordBatch' object has no attribute '_assert_cpu'
```

Reply:
Ugh, ignore what I said. I thought you were asking about removing from `_Tabular`. I agree, I can remove from this from `Table`

Reply:
And can remove it from `RecordBatch` as well?

Reply:
Good catch! I don't know why I didn't think of that.